src/param-builder.ts

```diff
@@ -176,6 +185,26 @@ class ParamBuilder {
     return baseUrl ? `${baseUrl}/${finalPath}` : `/${finalPath}`;
   }
 
+  /**
+   * Get the extension for the file. Checks the specified format and the target image.
+   *
+   * @param path The path to the target image, e.g. `https://example.com/foo.png`
+   * @returns An extension if found, or an empty string
+   */
+  private getExtention(path: string): string {
+    let extension = '';
+
+    const formatString = this.modifiers.get('format');
+    if (formatString) {
+      extension = `.${formatString.split(':').pop()}`;
+    } else {
+      const parts = path.split('.');
+      if (parts.length) extension = `.${parts.pop()}`;
+    }
+
+    return extension;
```

Comment:
I would not add the default/fallback extension since a file might not necessarily have an extension to begin with. 

```suggestion
    const formatString = this.modifiers.get('format');
    if(!formatString) return ''; 
    return `.${formatString.split(':').pop()}`;
```